src/main/java/eu/execom/hawaii/service/RequestService.java

```
@@ -136,52 +136,65 @@ public Request create(Request request) {
+          applyRequest(request, true);
+        } else if (!isUserRequestApprover(authUser, user) && isExistAsApproved(request.getId())) {
+          request.setRequestStatus(RequestStatus.CANCELLATION_PENDING);
+          emailService.createEmailAndSendForApproval(request);
+        }
+        break;
+      default:
+        log.info("Request with status {} will be saved without changing allowance for user with email {}",
+            request.getRequestStatus(), request.getUser().getEmail());
     }
 
     return requestRepository.save(request);
   }
 
-  private void checkIsApproverUserTeamApprover(User approver, User requestUser) {
-    if (requestUser.getTeam()
-                   .getTeamApprovers()
-                   .stream()
-                   .noneMatch(teamApprover -> teamApprover.getId().equals(approver.getId()))) {
-      log.error("Approver not authorized to approve this request for user with email: {}", requestUser.getEmail());
-      throw new NotAuthorizedApprovalExeception();
-    }
+  private boolean isUserRequestApprover(User approver, User requestUser) {
+    return requestUser.getTeam()
+                      .getTeamApprovers()
+                      .stream()
+                      .anyMatch(teamApprover -> teamApprover.getId().equals(approver.getId()));
   }
 
-  private boolean shouldApplyRequest(Request request) {
-    var isRequestExistingAsApproved = false;
-    if (request.getId() != null) {
-      var existingRequest = requestRepository.getOne(request.getId());
-      isRequestExistingAsApproved = request.getRequestStatus().equals(existingRequest.getRequestStatus());
-    }
-    var isRequestApproved = RequestStatus.APPROVED.equals(request.getRequestStatus());
-    var isRequestCanceled = RequestStatus.CANCELED.equals(request.getRequestStatus());
-    return isRequestApproved || (isRequestCanceled && isRequestExistingAsApproved);
+  private boolean isExistAsApproved(Long requestId) {
```

In my opinion this naming could be simplified to just `isApproved` and also the other one could be `isCancelled`. Since this method doesn't directly check if the request exists but just checks the status of it.

---

This isn`t checking from incoming data to be `isApproved` this checks current status from the database with incoming request status. For me, this looks more describable for the given situation.

---

Stefan has right, since it will not return false if the request does not exist, it will rather throw an exception for that case. See also my comment on the same topic.